main.cpp

```diff
@@ -442,97 +429,110 @@ struct ip_set
 	std::unordered_set<decltype(extract_key(std::declval<Address>()))> m_ips;
 };
 
+struct queued_node_t
+{
+	udp::endpoint ep;
+	int sock_idx;
+};
+
+template <typename Address>
 struct ping_queue_t
```

Comment:
while working on this increasingly more fields started having an IPv4 and IPv6 version, so making it a template made that a lot cleaner. I also moved in the queue entry type as an inner class